### ISSUE TITLE
[REVIEW]: Analysing 24-hour behaviour sequence data with an Rshiny application

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2282]
**Submitting author:** @jcolomb (<a href="http://orcid.org/0000-0002-3127-5520">Colomb, J.</a>)
**Repository:** <a href="https://github.com/jcolomb/HCS_analysis" target ="_blank">https://github.com/jcolomb/HCS_analysis</a>
**Version:** v0.1.3
**Editor:** @danielskatz
**Reviewer:** @mikerspencer, @aj2duncan
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/ecdda22f808463b29e7808b114c1fd01/status.svg)](https://joss.theoj.org/papers/ecdda22f808463b29e7808b114c1fd01)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/ecdda22f808463b29e7808b114c1fd01"><img src="https://joss.theoj.org/papers/ecdda22f808463b29e7808b114c1fd01/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/ecdda22f808463b29e7808b114c1fd01/status.svg)](https://joss.theoj.org/papers/ecdda22f808463b29e7808b114c1fd01)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/jcolomb/HCS_analysis" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@mikerspencer & @aj2duncan, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @danielskatz know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jcolomb/HCS_analysis">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@jcolomb) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/jcolomb/HCS_analysis">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@jcolomb) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @mikerspencer, @aj2duncan it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02282/joss.02282/10.21105.joss.02282.pdf)

**Comment by danielskatz** [comment]
👋 @mikerspencer & @aj2duncan - thanks again for agreeing to review this submission. Please be sure to read the comments above, and let me know if you have any questions.  Basically, your job is to check the article proof and repository and check items off your checklist above.

If you see small problems that need to be discussed, feel free to discuss them here.  But if you can, create a new issue in the <a href="https://github.com/jcolomb/HCS_analysis" target="_blank">target repository</a> and link to this review thread in that issue to create corresponding breadcrumb trail here.

I look forward to seeing how this review goes!

**Comment by mikerspencer** [comment]
@danielskatz to be clear - @aj2duncan and I work together a little. Reviewers working together isn't specifically mentioned in the conflict of interest, but I feel it's best to be transparent.

**Comment by danielskatz** [comment]
Thanks for letting me know. This isn't 100% ideal, but given that you both likely have different backgrounds and experiences, and we're focusing on a checklist-driven process, it should be ok. Let's see how it goes.

**Comment by aj2duncan** [comment]
Thanks @mikerspencer and apologies @danielskatz, I should have raised this too. I'm happy to provide more detail on this if needed and I'm pretty sure @mikerspencer would be too.

**Comment by danielskatz** [comment]
👋 @jcolomb - it looks like there's an issue blocking @mikerspencer ..

 @aj2duncan & @mikerspencer - how are your reviews coming along otherwise?

**Comment by aj2duncan** [comment]
@danielskatz now @jcolomb has [confirmed](https://github.com/jcolomb/HCS_analysis/issues/18) a particular part of the installation isn't necessary I will push on with the review.

**Comment by danielskatz** [comment]
Thanks - please make sure the documentation matches what you learn

**Comment by danielskatz** [comment]
👋 @aj2duncan & @mikerspencer - how are your reviews doing?

**Comment by aj2duncan** [comment]
@jcolomb quick query - In the shiny app and article you make reference to the 18 Berlin categories. Should this have a reference?

**Comment by jcolomb** [comment]
no, this is a "in-house" categorisation that was (so far) not published.

**Comment by aj2duncan** [comment]
> no, this is a "in-house" categorisation that was (so far) not published.

Thanks for the clarification.

**Comment by danielskatz** [comment]
👋 @aj2duncan & @mikerspencer - I'm just checking on your reviews and their progress again...

**Comment by aj2duncan** [comment]
@danielskatz everything seems to be moving along ok for me.

**Comment by mikerspencer** [comment]
@danielskatz can I check the scope of the review? How involved should we be as reviewers? I think testing should be done prior to review, but realise that's a grey area on problems picked up in the review.

**Comment by danielskatz** [comment]
@mikerspencer  - I don't think I understand what you are asking. You should do as much as possible to verify the criteria in the checklist, subject to reasonable effort, and then report what problems you come across so that the submitter can respond.  I'm confused when you say "I think testing should be done prior to review"  There should be some tests provided by the submitter (either manual or via some script or something else), and part of your job is to **verify** that they work and that you think they are reasonable, given the software, and the [JOSS documentation about testing](https://joss.readthedocs.io/en/latest/review_criteria.html#tests)

Let me know if this helps, and if not, please ask me again.

**Comment by mikerspencer** [comment]
Thanks @danielskatz your comment on verification is helpful. I think working on this issue (https://github.com/jcolomb/HCS_analysis/issues/18) is becoming more like co-creation. Keen to have your experienced view!

**Comment by danielskatz** [comment]
Well, ideally everything should work for the reviewer when they first try it, but if that was guaranteed, we wouldn't need any reviews :)

So, yes, sometimes the reviewers will work with the author to test things and this can be somewhat iterative.  If it's too much work, you can ask the author to go off and test with colleagues and come back when they are fairly sure the tests will pass for the reviewers, but sometimes, there is something tricky that does need to be worked out between reviewers and authors too.

Thanks for your help in doing this, and getting this far.

**Comment by mikerspencer** [comment]
No problem, thanks for clarifying - I appreciate it's a fine line!

@aj2duncan has done most of the work so far I think.

**Comment by danielskatz** [comment]
👋 @aj2duncan & @mikerspencer - As we're now almost 2 months into the review process, can you each please tell me:
1. Are you waiting on @jcolomb for anything?
2. If so, for what?
3. if not, when do you expect your review to be complete?

**Comment by aj2duncan** [comment]
Hi @danielskatz,

I'm waiting on jcolomb/HCS_analysis#20 and then I've got a couple of documentation things to check. I hope to have this done over the weekend.

**Comment by aj2duncan** [comment]
@danielskatz question for you. 

I think a statement of need is missing from the github repo but is present in the paper and @jcolomb has linked the paper from the first section of the README. Could you please comment on whether anything additional is required? 

Thanks.

**Comment by danielskatz** [comment]
> @danielskatz question for you.
> 
> I think a statement of need is missing from the github repo but is present in the paper and @jcolomb has linked the paper from the first section of the README. Could you please comment on whether anything additional is required?
> 
> Thanks.

It would probably be good to include some of this statement in the README directly.

**Comment by danielskatz** [comment]
👋 @jcolomb - how are you coming on the open issues?

**Comment by jcolomb** [comment]
I need some extra time. Sorry I can't be more reactive on this and thank you for the reminder !

**Comment by danielskatz** [comment]
👋 @jcolomb - another checkin - how are you coming on the open issues?

**Comment by jcolomb** [comment]
I am on it, probably get something showable this week.

**Comment by jcolomb** [comment]
hopefully done with the difficult parts, hope to finish tomorrow.

**Comment by jcolomb** [comment]
I am basically done, lots of debug and extension for online data in mbr format, documentation in readme and in metadata_information/readme were reworked quite a bit.
Changes are in the https://github.com/jcolomb/HCS_analysis/tree/joss_reviewanswers branch, and comment may be done in the pull request https://github.com/jcolomb/HCS_analysis/pull/27

Thank you a lot for all the feedback.

**Comment by danielskatz** [comment]
@aj2duncan & @mikerspencer - back over to you

**Comment by aj2duncan** [comment]
Thanks @jcolomb and @danielskatz, will look at this again as soon as I can.

**Comment by mikerspencer** [comment]
Cheers!

**Comment by danielskatz** [comment]
> Cheers!

Does that translate into American as "I'll get back to my review soon"? 🙂

**Comment by aj2duncan** [comment]
Hi @danielskatz, just to let you know that although there are a couple of small things to be ironed out I'll be done once we close jcolomb/HCS_analysis#20

**Comment by aj2duncan** [comment]
@danielskatz I think I've now finished my review. @jcolomb has dealt with all of my concerns. Thanks.

**Comment by danielskatz** [comment]
Thanks @aj2duncan!

**Comment by danielskatz** [comment]
👋 @mikerspencer - How is you review coming?  That's what we need to move forward, either to accept or to figure out what needs to be changed.

**Comment by danielskatz** [comment]
👋 @mikerspencer - How is your review coming?  Can you give us some idea when you might expect to complete it?

**Comment by mikerspencer** [comment]
Hi @danielskatz, apologies for the radio silence. I'm in the last two weeks of my current job, so things are a little frantic finishing tasks and tidying loose ends.

I will have more time from the 12th Oct, are you OK to wait until then?

**Comment by danielskatz** [comment]
Ok, we can wait until then (since trying to bring in another reviewer would likely be equally as long)

**Comment by danielskatz** [comment]
@whedon reminder @mikerspencer in 17 days

**Comment by whedon** [comment]
I'm sorry human, I don't understand that. You can see what commands I support by typing:

```
@whedon commands
```

**Comment by danielskatz** [comment]
whedon remind @mikerspencer in 17 days

**Comment by danielskatz** [comment]
@whedon remind @mikerspencer in 17 days

**Comment by whedon** [comment]
Reminder set for @mikerspencer in 17 days

**Comment by mikerspencer** [comment]
Many thanks - unfortunately life has overtaken me this time!

**Comment by whedon** [comment]
:wave: @mikerspencer, please update us on how your review is going.

**Comment by mikerspencer** [comment]
Hi, thanks for the check in. Should be finished tomorrow.

**Comment by mikerspencer** [comment]
I've paused my review while I wait for this issue to be fixed: https://github.com/jcolomb/HCS_analysis/issues/30.

**Comment by danielskatz** [comment]
@jcolomb - any update on this issue ⬆️ ?

**Comment by jcolomb** [comment]
It was fixed (I hope), sorry forgot to mention it here.

**Comment by danielskatz** [comment]
👋 @mikerspencer  - see https://github.com/jcolomb/HCS_analysis/issues/30#issuecomment-718730338

**Comment by danielskatz** [comment]
👋 @mikerspencer - any updates? This review has been going many months, and it would be really good to finish it soon.

I see a bunch of boxes checked, but some not checked. Are the unchecked boxes thing that you just haven't been able to check on yet, or are they items where there is work needed by the submitter, or discussion needed? If so, please say what these blocking issues are.

Otherwise, can you tell me when you think you will be done with your review?

**Comment by mikerspencer** [comment]
Hi @danielskatz - I agree, it's been a long time. The time I had available when I accepted the review has long gone and it's a shame the work wasn't ready when it was submitted.

My worry and opinion is the code base is extensive (1000s lines) with little obvious testing. I can run the apps as described in the docs, but it's not possible to validate the output which makes me uncomfortable signing the review off. The problem of verification is compounded by the software performing many tasks simultaneously (from a users view), which makes it hard to check whether data retrieval/cleaning/management/analysis/visualisation is happening as expected.

These are large and systematic problems to overcome. I'm aware dealing with them will not be a quick process, but I think it's something that needs to happen to pass review. An alternative, if the author is primarily interested in a citable record of the code is to link the Github repo to Zenodo (or Figshare/etc.).

**Comment by jcolomb** [comment]
The software already has a doi via Zenodo (since January 2018), my objective to send it here instead of writing a normal paper (as I did for the CeTrAn software years ago https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0042247, and as I had planned to do in 2018 -I actually had a manuscript ready as one can guess from some files in the repo-) was and still is to make sure the code has no technical flaws, is doing what I think it does, and is enough documented to allow contributions from other researchers. I am not a trained RSE and I take it also as a learning experience.

I am very sorry packrat was not working (anymore) when I submitted the software to JOSS and that it took so long to find a working solution. On the other hand, I think it is difficult to know what is "ready" for being peer reviewed, as software peer review is such a young field. 

I am not sure to understand what you call "testing", and thought providing a real test dataset would be sufficient. "Verifications" and "testing" were done on the go (especially by having different datasets that had different problems) and there is indeed no extra code for it, or fabricated data. However, using the master_noshiny.r file as a start, one can do the tasks step by step, and the different steps are creating different variables one can access if the whole code is run. That is how the code was created, worked and debugged... 

Are there specific things I can provide to overcome the "large and systematic problems" you mentioned and help with the "validation of the output" ?

**Comment by mikerspencer** [comment]
This chapter from Wickham's R Packages books is a really good grounding in testing: https://r-pkgs.org/tests.html.

Like you, I usually rely on informal testing when writing code. But this makes it very difficult for others to review our work. For your own purposes using a test dataset is perfect, but for a reviewer we don't know what the output to be. Writing formal tests (see link above) makes expectations explicit.

I completely agree and sympathise about learning and training for software development - academia is very slow to adopt or encourage better practice.

**Comment by jcolomb** [comment]
Thanks, it is indeed some eyes opening: "Always write a test when you discover a bug.".

I will need to learn how to use testing in practice and go through the code again. Indeed, probably a long process.
@danielskatz : I foresee again a long pause in the review process, is it ok to pause the review until I am done with this, probably a couple of months, if schools are not closing again ?

**Comment by danielskatz** [comment]
Hi @mikerspencer and @jcolomb  - I think there's some confusion here between best practices and what's required for JOSS.  I really appreciate the discussion, which I think will certainly improve the code, but the question here is if it currently meets the minimum required for JOSS publication.

For example, the testing criterion for JOSS is
> Automated tests: Are there automated tests or manual steps described so that the functionality of the software can be verified?

Full unit testing and CI is not needed to pass this, though it would be great if that the code had that.  What's needed at a minimum is a description of the manual steps that someone who installs the code could use to verify that it generally was installed successful and can achieve some set of known good results.

This means that the comment from @mikerspencer 
> it's not possible to validate the output which makes me uncomfortable signing the review off. The problem of verification is compounded by the software performing many tasks simultaneously (from a users view), which makes it hard to check whether data retrieval/cleaning/management/analysis/visualisation is happening as expected.

is a problem that needs to be resolved.

@jcolomb - maybe you can propose a set of changes that would allow @mikerspencer to say that the software meets the *minimum* for the JOSS criteria that are still unchecked, and then we can see how much work and how many changes that would be.  We could then either pause this while that work happens, and if @mikerspencer is not available then, I will find another reviewer, or withdraw the submission and start again after that.

**Comment by mikerspencer** [comment]
Thanks Daniel, I think that's a pragmatic way forward.

**Comment by danielskatz** [comment]
Thanks - then we will wait for @jcolomb to respond, hopefully soon

**Comment by jcolomb** [comment]
I am on it! I will get a extra code meant for review.

**Comment by danielskatz** [comment]
Hi @jcolomb - just checking to see how this is coming

**Comment by jcolomb** [comment]
hi, I had some urgent matter to attend to, I hope to get the svm "reviewable" after the Christmas break.

**Comment by danielskatz** [comment]
👋 @jcolomb - I hope you had a good break, and want to check in on this again so that we can get it done...

**Comment by jcolomb** [comment]
thanks, happy new year. We are back in hardcore lockdown (i.e. schools are closed) so it might be delayed a bit more, I will give updates weekly. Sorry again for the delays, and thank you for your patience. @mikerspencer thank you again for the tips, I did the first part if you have time to have a look, "only" the SVM analysis has not been tested. You may want to review here: https://github.com/jcolomb/HCS_analysis/pull/35/files#diff-96933a5a509b8888527a6e745aa92c169f5394a525339682fa1cec926937cf5e

**Comment by mikerspencer** [comment]
Essentially lockdown here too. Also juggling childcare and work - fun times!

**Comment by danielskatz** [comment]
@jcolomb - since it's been two weeks without an update, I'm checking in

**Comment by jcolomb** [comment]
thanks, I nearly finished offline, still need to do a couple of  last checks and then next push should do it !

**Comment by danielskatz** [comment]
Sorry I missed this somehow and am just catching up

**Comment by danielskatz** [comment]
👋 @mikerspencer - when you get a chance, can you take a look at [the new PR](https://github.com/jcolomb/HCS_analysis/pull/35) from @jcolomb?

**Comment by danielskatz** [comment]
@mikerspencer - ping...  ☝️

**Comment by danielskatz** [comment]
@mikerspencer - ping... ☝️  (I know things are tough these days, but I think we're close to finishing this, after a long effort)

**Comment by mikerspencer** [comment]
Hi all, apologies for the delay. I'd missed these in the rush of home schooling. Normal service is back this week so I'll get it finished off.

**Comment by danielskatz** [comment]
Hi @mikerspencer - just checking to see how the week went...

**Comment by danielskatz** [comment]
👋 @mikerspencer - checking in again

**Comment by danielskatz** [comment]
👋 @mikerspencer - checking in again

**Comment by danielskatz** [comment]
👋 @jcolomb - sorry for the delays here.  @mikerspencer and I have chatted briefly offline so that I could better understand his concerns.  Then I asked a couple of other JOSS editors to take a look as well.  Our opinion is that this is not ready to be published at this point.  @karthik has some specific suggestions, which I've asked him to add here, basically reinforcing what @mikerspencer was uncomfortable about

**Comment by karthik** [comment]
👋 @jcolomb 
Upon inspection of the submissions, it is not really clear what you are trying to accomplish or how someone else will reuse or benefit from this analysis package. The installation is really complicated, dependencies are unclear (for example, one should never directly import tidyverse in a package setting, but call on specific packages and functions as needed. Otherwise you are importing a massive kitchen sink).

Suggestions for refactoring:
- Write this as a regular R package with exported functions so someone can use it in a script of their own
- Include the shiny app inside the package, so it can serve as a demo.
- The background is not clear enough. Please rewrite the readme so it is written for someone who doesn't already work in your lab.
- I think more work is needed (as you say *We hope to make the software flexible enough to analyse data coming for other similar setups.*) before this would be general enough for a JOSS audience.

Best,
Karthik

**Comment by danielskatz** [comment]
@jcolomb - given the comments from @mikerspencer and @karthik, I'm going to reject this, but we would welcome a resubmission of a future version that takes these comments into account.  I think this would be sufficiently different that it doesn't make sense to pause this and restart it at that time.

**Comment by danielskatz** [comment]
@whedon reject

**Comment by whedon** [comment]
Paper rejected.

**Comment by jcolomb** [comment]
Dear @danielskatz and @karthik, thank you for taking that much time to discuss this case.

**Comment by mikerspencer** [comment]
Hi @jcolomb,

Some comments to help with subsequent work:

* Have a look at the Wickham & Bryan book on R packages (https://r-pkgs.org/). It's got some really helpful material on structuring work and testing.
* Try and break down work/repos into sections, this will make it easier for others to follow logic and find the parts they need.
* Focus the repo on a task and remove code that no longer contributes to the goal. This will likely evolve over time.

Best of luck with future work,
Mike
